enterprise/server/ociregistry/ociregistry.go

```diff
@@ -167,6 +182,16 @@ func (r *registry) handleBlobsOrManifestsRequest(ctx context.Context, w http.Res
 		return
 	}
 
+	bsClient := r.env.GetByteStreamClient()
+	acClient := r.env.GetActionCacheClient()
+	if identifierIsDigest {
+		writeBody := inreq.Method == http.MethodGet
+		err := fetchBlobOrManifestFromCache(ctx, w, bsClient, acClient, ref, blobsOrManifests, writeBody)
+		if err == nil {
```

Comment:
Should log the error here if it's not a NotFound error.

Reply:
Added logging and now also respond HTTP 404 if there's an error other than `NotFound`. It's possible that `fetchBlobOrManifestFromCache` has already written to the `ResponseWriter`.